api/src/main/java/io/opentelemetry/trace/SpanContext.java

```diff
@@ -156,40 +166,36 @@ public boolean equals(@Nullable Object obj) {
     }
 
     SpanContext that = (SpanContext) obj;
-    return traceId.equals(that.traceId)
-        && spanId.equals(that.spanId)
-        && traceFlags.equals(that.traceFlags)
-        && isRemote == that.isRemote;
+    return getTraceId().equals(that.getTraceId())
+        && getSpanId().equals(that.getSpanId())
+        && getTraceFlags().equals(that.getTraceFlags())
+        && isRemote() == that.isRemote();
   }
 
+  /**
+   * Note: This has a custom equals/hashcode implementation. The hashcode() method does not include
+   * the {@code tracestate} or the {@code remote} flag.
+   *
+   * <p>TODO: is this intentional? If it is, it should be documented as to *why* this is the case.
+   */
   @Override
-  public int hashCode() {
-    return Arrays.hashCode(new Object[] {traceId, spanId, traceFlags});
+  public final int hashCode() {
+    return Arrays.hashCode(new Object[] {getTraceId(), getSpanId(), getTraceFlags()});
   }
 
-  @Override
-  public String toString() {
-    return "SpanContext{traceId="
-        + traceId
-        + ", spanId="
-        + spanId
-        + ", traceFlags="
-        + traceFlags
-        + ", isRemote="
-        + isRemote
-        + "}";
-  }
+  /** A {@code Builder} for a {@link SpanContext}. */
+  @AutoValue.Builder
+  abstract static class Builder {
```

Comment:
Let's not be that Java :)), use just a simple Create method and avoid creating a builder every time we create a SpanContext.

Reply:
🤣  Ok, will do.

Reply:
done